src/app/services/requests.service.ts

```diff
@@ -1,37 +1,52 @@
-import { Injectable } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
 import { Http, Response, Headers } from '@angular/http';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/observable/throw';
+import { ConfigurationService } from './configuraion.service';
+import { DataPathUtils } from '../utils/dataPath.utils';
 
 @Injectable()
 export class RequestsService {
-
-  constructor(public http: Http) {}
+  private errorMessageDataPaths: string[] = [];
+
+  constructor(public http: Http,
+              @Inject("DataPathUtils") private readonly dataPathUtils: DataPathUtils,
+              @Inject("ConfigurationService") configurationService: ConfigurationService) {
+    configurationService.getConfiguration().subscribe(configuration => {
+      if (configuration.errorMessageDataPath) {
+        if (Array.isArray(configuration.errorMessageDataPath)) {
+          this.errorMessageDataPaths = configuration.errorMessageDataPath
+        } else {
+          this.errorMessageDataPaths = [configuration.errorMessageDataPath];
+        }
+      }
+    });
+  }
 
   public get(url, headers = null, queryParams = null) {
     return this.http.get(this.buildUrl(url, queryParams), { headers: this.buildHeaders(headers) })
       .map(this.extractData)
-      .catch(this.handleError);
+      .catch(error => this.handleError(error));
```

Comment:
I'm sorry, but what's the difference?

Reply:
If I don't make the call within an arrow function, `this` is undefined when `handleError` is executing.

Here is some discussion about what's going on in the underlying javascript: https://stackoverflow.com/questions/20279484/how-to-access-the-correct-this-inside-a-callback

Reply:
Got it.